package-system/expat/Findexpat.cmake

```diff
@@ -0,0 +1,64 @@
+#
+# Copyright (c) Contributors to the Open 3D Engine Project.
+# For complete copyright and license terms please see the LICENSE at the root of this distribution.
+# 
+# SPDX-License-Identifier: Apache-2.0 OR MIT
+#
+#
+
+# the following is like an include guard:
+if (TARGET 3rdParty::expat)
+    return()
+endif()
+
+# Even though expat itself exports it as lowercase expat, older cmake (and cmake's built-in targets)
+# expect uppercase.  So we define both, for backwards compat:
+
+if (WIN32)
+    # on windows, expat adds the nonstandard 'lib' prefix and MD, dMD suffixes for 
+    # Multighreaded Dynamic CRT and debug Multithreaded Dynamic CRT
+    # We don't use the debug version since its a pure C library with no C++ and thus will
+    # not have an ITERATOR_DEBUG_LEVEL conflict
+    set(PREFIX_TO_USE "lib")
+    set(SUFFIX_TO_USE "MD.lib")
+else()
+    # on other platforms its just standard prefixes and suffix
+    set(PREFIX_TO_USE ${CMAKE_STATIC_LIBRARY_PREFIX})
+    set(SUFFIX_TO_USE ${CMAKE_STATIC_LIBRARY_SUFFIX})
+endif()
+
+set(EXPAT_LIBRARY ${CMAKE_CURRENT_LIST_DIR}/expat/lib/${PREFIX_TO_USE}expat${SUFFIX_TO_USE})
+set(expat_LIBRARY ${EXPAT_LIBRARY})
+
+set(EXPAT_INCLUDE_DIR ${CMAKE_CURRENT_LIST_DIR}/expat/include)
+set(expat_INCLUDE_DIR ${EXPAT_INCLUDE_DIR})
+
+set(EXPAT_FOUND TRUE)
+set(expat_FOUND TRUE)
+
+add_library(expat::expat STATIC IMPORTED GLOBAL)
+set_target_properties(expat::expat PROPERTIES IMPORTED_LINK_INTERFACE_LANGUAGES "C")
+set_target_properties(expat::expat PROPERTIES IMPORTED_LOCATION ${EXPAT_LIBRARY})
+target_compile_definitions(expat::expat INTERFACE -DXML_STATIC)
```

Comment:
this define seems quite generic, the `-D` is also not needed (and is preferred to not have it in case some compiler passes defines in a different way)
If we are not using that `XML_STATIC` define, I would suggest to remove this line

Reply:
Ouch, yeah, its absolutely needed.  ITs what causes EXPAT's headers (ie, expat.h) to not try to dll import.

The weird thing is, this is working on every compiler and every platform we use, Jenkins can link and build them, etc... so CMake must either understand -DXML_STATIC  or -DXML_STATIC is working across every platform (?).

Should I hold out and rev the package?  Or ticket it for next time we rev expat?  Any thoughts?

Reply:
Unfortunately abotu the name of the define, just something to be careful about.

Yes, CMake, removes the `-D`: https://cmake.org/cmake/help/latest/command/target_compile_definitions.html
Fine to do it in a future change, is totally minor/cosmetic

Reply:
whew, saved by cmake maintainers / developers :)